kernel/sys/tasks.cpp

```diff
@@ -98,6 +118,39 @@ static inline uint64_t _get_cpu_time_ns()
     return (__rdtsc()) / _instr_per_ns;
 }
 
+static void _print_task(const px_task_t *task)
+{
+    px_rs232_printf("%s is %s\n", task->name, _state_names[task->state]);
+}
+
+#ifdef DEBUG
+#define TASK_ACTION(action, task) do { px_rs232_print(action " "); _print_task(task); } while(0)
```

Comment:
Do we really need the `do { ... } while(0)` loop here?

Reply:
Not really, but it's the "proper" thing to do. See https://stackoverflow.com/questions/154136/why-use-apparently-meaningless-do-while-and-if-else-statements-in-macros

Reply:
I still don't like it but it makes sense.